### PR TITLE
fix spinner staying after error

### PR DESCRIPTION
**Pull request by Ortega-Dan**
# Basics
Ticket: https://dol-ewf.atlassian.net/browse/DSG-2445

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [x] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

This avoids the form-selection-rules subcomponent's spinner to keep in front of the screen on any failure loading contents.

## Screenshot of the affected changes

Video of both before and after:
https://github.com/Nuvalence/dsgov-web/assets/24926168/7b0431b9-20a4-433c-8d85-b9ed3a851ec1

## Checklist

- [ ] I have updated the documentation accordingly
- [x] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by davidyoung77 — APPROVED** [review]
Looks good

File: libs/agency/feature/admin/src/lib/components/form-selection-rules/form-selection-rules.component.ts [hunk 1]
```diff
@@ -123,36 +119,25 @@ export class FormSelectionRulesComponent implements OnInit {
...
     );
 
-    this.loadForms$ = this._formConfigurationService.getFormConfigurations$(this.transactionDefinition.key).pipe(
-      catchError(_ => {
-        this._nuverialSnackBarService.notifyApplicationError();
-
-        return EMPTY;
-      }),
-    );
+    this.loadForms$ = this._formConfigurationService.getFormConfigurations$(this.transactionDefinition.key).pipe();
```
Comment (thechechogomez) [inline libs/agency/feature/admin/src/lib/components/form-selection-rules/form-selection-rules.component.ts:124]: nit: I think we should remove the pipe as it holds nothing in it.

**Review by thechechogomez — APPROVED** [review]
Looks great. A small nitpick

File: libs/agency/feature/admin/src/lib/components/form-selection-rules/form-selection-rules.component.ts (hunk 1 above)
Comment (Ortega-Dan) [reply libs/agency/feature/admin/src/lib/components/form-selection-rules/form-selection-rules.component.ts]: Thank @thechechogomez . I applied it and rebased to latest main to be able to merge. Would you please re-approve ?

**Review by thechechogomez — APPROVED** [review]

**Pull request merged**